rclpy/rclpy/lifecycle/node.py

```diff
@@ -0,0 +1,392 @@
+            except _rclpy.RCLError:
+                resp.success = False
+                return resp
+        cb_return_code = self.__change_state(transition_id)
+        resp.success = cb_return_code == TransitionCallbackReturn.SUCCESS
+        return resp
+
+    def __on_get_state(
+        self,
+        req: lifecycle_msgs.srv.GetState.Request,
+        resp: lifecycle_msgs.srv.GetState.Response
+    ):
+        self.__check_is_initialized()
+        resp.current_state.id, resp.current_state.label = self._state_machine.current_state
+        return resp
+
+    def __on_get_available_states(
+        self,
+        req: lifecycle_msgs.srv.GetAvailableStates.Request,
+        resp: lifecycle_msgs.srv.GetAvailableStates.Response
+    ):
+        self.__check_is_initialized()
+        for state_id, label in self._state_machine.available_states:
+            resp.available_states.append(lifecycle_msgs.msg.State(id=state_id, label=label))
+        return resp
+
+    def __on_get_available_transitions(
+        self,
+        req: lifecycle_msgs.srv.GetAvailableTransitions.Request,
+        resp: lifecycle_msgs.srv.GetAvailableTransitions.Response
+    ):
+        self.__check_is_initialized()
+        for transition_description in self._state_machine.available_transitions:
+            transition_id, transition_label, start_id, start_label, goal_id, goal_label = \
+                transition_description
+            item = lifecycle_msgs.msg.TransitionDescription()
+            item.transition.id = transition_id
+            item.transition.label = transition_label
+            item.start_state.id = start_id
+            item.start_state.label = start_label
+            item.goal_state.id = goal_id
+            item.goal_state.label = goal_label
+            resp.available_transitions.append(item)
+        return resp
+
+    def __on_get_transition_graph(
+        self,
+        req: lifecycle_msgs.srv.GetAvailableTransitions.Request,
+        resp: lifecycle_msgs.srv.GetAvailableTransitions.Response
+    ):
+        self.__check_is_initialized()
+        for transition_description in self._state_machine.transition_graph:
+            transition_id, transition_label, start_id, start_label, goal_id, goal_label = \
+                transition_description
+            item = lifecycle_msgs.msg.TransitionDescription()
+            item.transition.id = transition_id
+            item.transition.label = transition_label
+            item.start_state.id = start_id
+            item.start_state.label = start_label
+            item.goal_state.id = goal_id
+            item.goal_state.label = goal_label
+            resp.available_transitions.append(item)
+        return resp
+
+
+class LifecycleNode(LifecycleNodeMixin, Node):
+    """
+    A ROS 2 managed node.
+
+    This class extends Node with the methods provided by LifecycleNodeMixin.
+    Methods in LifecycleNodeMixin overridde the ones in Node.
```

Comment:
```suggestion
    Methods in LifecycleNodeMixin override the ones in Node.
```

Reply:
Fixed in https://github.com/ros2/rclpy/pull/865/commits/324d4ef39fcba3216c05e4a3c089af009fd1c174